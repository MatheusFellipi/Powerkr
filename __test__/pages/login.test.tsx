import { render } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import Login from "../../src/app/index";
import config from "../../tamagui.config";

describe("<Login />", () => {
  it("renders correctly", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <Login />
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
