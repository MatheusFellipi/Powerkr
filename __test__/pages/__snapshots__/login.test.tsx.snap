// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "height": 1334,
        "paddingTop": 130,
        "width": 750,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <SvgMock />
      <View
        style={
          {
            "flexDirection": "row",
            "marginTop": 24,
          }
        }
      >
        <SvgMock />
        <View
          pointerEvents="none"
          style={
            {
              "height": 24,
              "minHeight": 24,
              "minWidth": 24,
              "width": 24,
            }
          }
        />
        <SvgMock />
      </View>
    </View>
    <View
      style={
        {
          "marginBottom": 24,
          "marginTop": 144,
          "paddingLeft": 24,
        }
      }
    >
      <Text
        fontFamily="Inter"
        style={
          {
            "color": "#fff",
            "fontFamily": "Inter",
            "fontSize": 30,
          }
        }
        suppressHighlighting={true}
      >
        Bem-vindo de volta!
      </Text>
      <Text
        fontFamily="Inter"
        style={
          {
            "color": "#CDD5DF",
            "fontFamily": "Inter",
            "fontSize": 14,
          }
        }
        suppressHighlighting={true}
      >
        Preparado para mais uma jornada?
      </Text>
    </View>
    <View
      onSubmit={[Function]}
      style={
        {
          "backgroundColor": "#182E43",
          "borderStyle": "solid",
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
          "borderWidth": 1,
          "height": 400,
          "paddingHorizontal": 20,
          "paddingTop": 20,
          "width": 750,
        }
      }
    >
      <View
        style={{}}
      >
        <Text
          cursor="default"
          fontFamily="$body"
          htmlFor="email"
          id=":r0:"
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "color": "#CDD5DF",
              "display": "flex",
              "fontFamily": "Inter",
              "fontSize": 14,
              "letterSpacing": 0,
              "lineHeight": 44,
            }
          }
          suppressHighlighting={true}
          userSelect="none"
        >
          Informações da conta
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#364152",
              "borderColor": "#4B5565",
              "borderRadius": 24,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            focusable={true}
            keyboardType="email-address"
            onChangeText={[Function]}
            outlineWidth={0}
            placeholder="Insira seu e-mail"
            placeholderTextColor="#545454"
            style={
              {
                "backgroundColor": "rgba(10,10,10,0)",
                "borderBottomColor": "rgba(10,10,10,0)",
                "borderBottomLeftRadius": 24,
                "borderBottomRightRadius": 24,
                "borderBottomWidth": 1,
                "borderLeftColor": "rgba(10,10,10,0)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgba(10,10,10,0)",
                "borderRightWidth": 1,
                "borderStyle": "solid",
                "borderTopColor": "rgba(10,10,10,0)",
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
                "borderTopWidth": 1,
                "color": "#fff",
                "fontFamily": "Inter",
                "fontSize": 14,
                "height": 44,
                "letterSpacing": 0,
                "minWidth": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
                "width": "90%",
              }
            }
          />
        </View>
      </View>
      <View
        style={{}}
      >
        <Text
          cursor="default"
          fontFamily="$body"
          htmlFor="password"
          id=":r1:"
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "color": "#CDD5DF",
              "display": "flex",
              "fontFamily": "Inter",
              "fontSize": 14,
              "letterSpacing": 0,
              "lineHeight": 44,
            }
          }
          suppressHighlighting={true}
          userSelect="none"
        >
          Senha
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#364152",
              "borderColor": "#4B5565",
              "borderRadius": 24,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            focusable={true}
            onChangeText={[Function]}
            outlineWidth={0}
            placeholder="Senha"
            placeholderTextColor="#545454"
            secureTextEntry={true}
            style={
              {
                "backgroundColor": "rgba(10,10,10,0)",
                "borderBottomColor": "rgba(10,10,10,0)",
                "borderBottomLeftRadius": 24,
                "borderBottomRightRadius": 24,
                "borderBottomWidth": 1,
                "borderLeftColor": "rgba(10,10,10,0)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgba(10,10,10,0)",
                "borderRightWidth": 1,
                "borderStyle": "solid",
                "borderTopColor": "rgba(10,10,10,0)",
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
                "borderTopWidth": 1,
                "color": "#fff",
                "fontFamily": "Inter",
                "fontSize": 14,
                "height": 44,
                "letterSpacing": 0,
                "minWidth": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
                "width": "90%",
              }
            }
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "paddingRight": 10,
                "width": "10%",
              }
            }
          >
            <SvgMock
              fill="#A0AEC0"
              height={25}
              width={25}
            />
          </View>
        </View>
      </View>
      <Text
        fontFamily="Inter"
        style={
          {
            "color": "#F7B519",
            "fontFamily": "Inter",
            "fontSize": 14,
            "marginBottom": 24,
            "marginTop": 24,
            "textAlign": "right",
          }
        }
        suppressHighlighting={true}
      >
        Esqueceu a senha?
      </Text>
      <View
        accessibilityRole="button"
        cursor="pointer"
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F7B519",
            "borderBottomColor": "#4B5565",
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderBottomWidth": 1,
            "borderLeftColor": "#4B5565",
            "borderLeftWidth": 1,
            "borderRightColor": "#4B5565",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "#4B5565",
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
            "borderTopWidth": 1,
            "color": "#000",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 44,
            "justifyContent": "center",
            "paddingLeft": 18,
            "paddingRight": 18,
            "width": "100%",
          }
        }
      >
        <Text
          cursor="pointer"
          fontFamily="InterBold"
          lineBreakMode="clip"
          numberOfLines={1}
          style={
            {
              "color": "#000",
              "flexGrow": 0,
              "flexShrink": 1,
              "fontFamily": "InterBold",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 23,
            }
          }
          suppressHighlighting={true}
          userSelect="none"
        >
          Entrar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
